### PR TITLE
Fix Wrong API Error Message

### PR DESCRIPTION
**Pull request by jumaallan**
Story: https://app.shortcut.com/smileid/story/13459/ios-fix-wrong-message-from-the-api

## Summary

A few sentences/bullet points about the changes

## Known Issues

Any shortcomings in your work. This may include corner cases not correctly handled or issues related
to but not within the scope of your PR. Design compromises should be discussed here if they were not
already discussed above.

## Test Instructions

Concise test instructions on how to verify that your feature works as intended. This should include
changes to the development environment (if applicable) and all commands needed to run your work.

## Screenshot

If applicable (e.g. UI changes), add screenshots to help explain your work.

### CODE REVIEW

**Comment by jumaallan** [comment]
Part of #230

File: Sources/SmileID/Classes/Networking/URLSession/URLSessionRestServiceClient.swift [hunk 1]
```diff
@@ -74,22 +74,33 @@ class URLSessionRestServiceClient: NSObject, RestServiceClient {
...
+                from: urlSessionResponse.data
+            ) {
+                throw SmileIDError.httpError((
                     urlSessionResponse.response as? HTTPURLResponse
-                )?.statusCode ?? 500,
-                urlSessionResponse.data
+                )?.statusCode ?? 500, httpError.error)
+            }
+            throw SmileIDError.httpError(
+                (urlSessionResponse.response as? HTTPURLResponse)?.statusCode ?? 500,
```
[the remark this answers is not in the document]
Comment (EricNdivo) [reply Sources/SmileID/Classes/Networking/URLSession/URLSessionRestServiceClient.swift:102]: Perhaps try throwing the original error instead of falling back to a default code. This might make it easier to handle edge cases without introducing misleading server errors where none exist.
